### ISSUE TITLE
`T of mixed|never` is a "more specific type" than `T of mixed`

### ISSUE THREAD

**Issue by someniatko** [vimeo/psalm#10377]
https://psalm.dev/r/4967ec67b9

This behavior broke Psalm checks for my lib: https://github.com/someniatko/php-result-type when I tried to update its automatic  checks from Psalm 4 to Psalm 5. IDK which version specifically changed the behavior. Personally to me it looks like a bug.

**Comment by someniatko** [comment]
> If you could explain your use case in a more practical example (why you need |never added in the @return), someone might be able to help you from there.

That's exactly why I have mentioned my library and the specific classes to look into!

**Comment by someniatko** [comment]
@kkmuffme So have you looked into the lib? https://github.com/someniatko/php-result-type/blob/master/src/None.php#L9

**Comment by klimick** [comment]
Real example with `Option<A>`: https://psalm.dev/r/5534e23785
With `mixed` typehints there is not issue: https://psalm.dev/r/fdb283b340

**Comment by someniatko** [comment]
> calling `return $else();` indeed does not return `mixed|never` but `mixed`

This fact is 100% correct! However, that is exactly what I am trying to explain in the issue from the beginning. `mixed|never` should be identical to `mixed`, the same as it is in non-templated example:

```
/**
 * @return mixed|never
 */
function mixed(mixed $something)
{
    return $something;
}
```

Since `never` is a subtype of any type, `mixed|never` is a subtype of `mixed`.

**Comment by someniatko** [comment]
@kkmuffme Yeah, I get your point, however this is somewhat narrowing the original "by-book" `never` interpretation. In the particular case of my lib, I need `never` as a bottom type, without the "exit" semantics whatsoever. It is unfortunately impossible to properly type the `Option` class and its subclasses for it to stay covariant, without using `never` in its "pure" bottom-type semantics.

**Comment by someniatko** [comment]
Okay, let me move myself to a more constructive realm :D

What do you suggest to use instead of `never` for my use-case?
